feat: add 2.13.0

vaibhavkhadilkar opened the pull request:
Updates to Meteor v2.13.0 using the following command --
./update.sh --meteor-version 2.13 --node-version 14.21.4

GeoffreyBooth commented:
There doesn’t appear to be a `node:14.21.4-alpine` Docker image.

vaibhavkhadilkar commented:
I will keep a lookout for when the image is released and report back.

GeoffreyBooth commented:
Node won’t be publishing a 14.21.4; Meteor started using a fork: https://blog.meteor.com/new-meteor-js-2-13-node-js-14-21-4-security-patch-and-blaze-2-7-1-release-60134947e4c

Meteor isn’t publishing prior versions, though, so you’ll need to add some logic that detects Node versions >= 14.21.4 < 18 and if so, use the Meteor fork of the Node Docker image.

vaibhavkhadilkar commented:
Got it. I will work on adding the appropriate logic to detect Node versions correctly.

One question: which of the Docker images listed [here](https://hub.docker.com/r/meteor/node/tags) would be an appropriate choice?

GeoffreyBooth commented:
> One question: which of the Docker images listed here would be an appropriate choice?

I guess either 14.21.4-alpine3.17 or 14.21.4-alpine, I'm not sure it matters.

vaibhavkhadilkar commented:
I don't see a 14.21.4-alpine, so I will proceed with 14.21.4-alpine3.17. Thanks.

vaibhavkhadilkar commented:
I've fixed the update.sh script with logic that uses Meteor's Node fork of the Docker image for Node versions >= 14.21.4 and < 18.0.0. Then, I've used the updated script to update files to Meteor v2.13 using the following command --
./update.sh --meteor-version 2.13 --node-version 14.21.4

Please review and let me know what you think.

GeoffreyBooth commented:
I think you have some bugs (see CI) but the general approach looks fine. You can test locally with `build.sh && test.sh`.

vaibhavkhadilkar commented:
Thank you for the feedback. I will work on the bugs and report back.

vaibhavkhadilkar commented:
I've fixed the bugs in the update.sh script and all local tests also pass.
I've also updated the test.sh script to use the same logic in update.sh to correctly display the Node version in the pass/fail text message.
Please review and let me know what you think. Thanks.

GeoffreyBooth commented on test.sh:
```diff
@@ -12,6 +12,10 @@ run_with_suppressed_output () {
 
+# Use printf to get appropriate version string for comparison
+get_version_string() {
+        printf "%02d%02d%02d" $(echo "$1" | tr '.' ' ');
```
```suggestion
	printf "%02d%02d%02d" $(echo "$1" | tr '.' ' ');
```

Rather than copying this into two files, can you please put it in https://github.com/disney/meteor-base/blob/main/support.sh? That’s the file with all the shared code.

vaibhavkhadilkar replied:
Thanks, I've moved the function to support.sh.

GeoffreyBooth commented on support.sh:
```diff
@@ -60,5 +60,12 @@ set_node_version() {
+
+# Use printf to get appropriate version string for comparison
+get_version_string() {
+        printf "%02d%02d%02d" $(echo "$1" | tr '.' ' ');
```
```suggestion
	printf "%02d%02d%02d" $(echo "$1" | tr '.' ' ');
```

GeoffreyBooth commented:
Thanks for your contribution!

vaibhavkhadilkar commented:
Thank you for your helpful feedback and guidance!
